tools/psa/release.py

```diff
@@ -0,0 +1,143 @@
+    '*psa-crypto_access_control': ['USE_PSA_TEST_PARTITIONS', 'USE_CRYPTO_ACL_TEST']
+}
+
+
+def get_mbed_official_psa_release():
+    def psa_backend(target_name):
+        return 'TFM' if 'TFM' in Target.get_target(target_name).labels else 'MBED_SPM'
+
+    psa_targets_release_dict = {
+        'TFM': [],
+        'MBED_SPM': []
+    }
+
+    psa_secure_targets = [t for t in TARGET_NAMES if Target.get_target(t).is_PSA_secure_target]
+    for t in psa_secure_targets:
+        psa_targets_release_dict[psa_backend(t)].append(
+            tuple(
+                [
+                    TARGET_MAP[t].name,
+                    TARGET_MAP[t].default_toolchain
+                ]
+            )
+        )
+
+    return psa_targets_release_dict
+
+
+def create_mbed_ignore(dir):
+    with open(os.path.join(dir, '.mbedignore'), 'w') as f:
+        f.write('*\n')
+
+
+def build_mbed_spm_platform(target, toolchain):
+    subprocess.call([
+        PYTHON_EXEC,
+        '-u',
+        TEST_PY_LOCATTION,
+        '--greentea',
+        '--profile', 'debug',
+        '-t', toolchain,
+        '-m', target,
+        '--source', ROOT,
+        '--build', os.path.join(ROOT, 'BUILD', 'tests', target),
+        '--test-spec', os.path.join(ROOT, 'BUILD', 'tests', target, 'test_spec.json'),
+        '--build-data', os.path.join(ROOT, 'BUILD', 'tests', target, 'build_data.json'),
+        '-n', MBED_PSA_TESTS
+    ])
+
+    subprocess.call([
+        PYTHON_EXEC, '-u', MAKE_PY_LOCATTION,
+        '-t', toolchain,
+        '-m', target,
+        '--profile', 'release',
+        '--source', ROOT,
+        '--build', os.path.join(ROOT, 'BUILD', target),
+        '--artifact-name', 'psa_release_1.0'
+    ])
+
+
+def tfm_test_defines(test):
+    return ['-D{}'.format(define) for define in TFM_TESTS[test]]
+
+
+def build_tfm_platform(target, toolchain):
+    for test in TFM_TESTS.keys():
+        subprocess.call([
+            PYTHON_EXEC,
+            '-u',
+            TEST_PY_LOCATTION,
+            '--greentea',
+            '--profile', 'debug',
+            '-t', toolchain,
+            '-m', target,
+            '--source', ROOT,
+            '--build', os.path.join(ROOT, 'BUILD', 'tests', target),
+            '--test-spec', os.path.join(ROOT, 'BUILD', 'tests', target, 'test_spec.json'),
+            '--build-data', os.path.join(ROOT, 'BUILD', 'tests', target, 'build_data.json'),
+            '--app-config', TFM_MBED_APP, '-n', test] + tfm_test_defines(test))
+
+    subprocess.call([
+        PYTHON_EXEC, '-u', MAKE_PY_LOCATTION,
+        '-t', toolchain,
+        '-m', target,
+        '--profile', 'release',
+        '--source', ROOT,
+        '--build', os.path.join(ROOT, 'BUILD', target),
+        '--app-config', TFM_MBED_APP
+    ])
+
+
+def main():
+    buid_dir = os.path.join(ROOT, 'BUILD')
+    if os.path.exists(buid_dir):
+        shutil.rmtree(buid_dir)
```

Comment:
You're always doing a clean build. This is probably really slow, and not needed.

Reply:
no...cleaning the environment in the beginning once and that's it

Reply:
Yes? every time you run the script it's a clean build. This script always does a clean build.

Reply:
just a precaution